src/v/kafka/client/client.h

```diff
@@ -96,10 +96,9 @@ class client {
     ss::future<typename std::invoke_result_t<
       Func>::api_type::response_type> dispatch(Func func) {
         return gated_retry_with_mitigation([this, func{std::move(func)}]() {
-            return _brokers.any().then(
-              [func{std::move(func)}](shared_broker_t broker) mutable {
```

Comment:
`kafka/client: Don't move func in retry`

... because it was a copy anyway?

Reply:
Clarified the commit message:
```
The capturing lambda is not mutable, so the move didn't occur.       

If the capturing lambda is made mutable, then func must also be
mutable, which is not desirable.
```